### ISSUE TITLE
Can't generate for very long

### ISSUE THREAD

**Issue by pkp24** [remsky/Kokoro-FastAPI#33]
I generated a bunch of short little test pieces and tried one that was maybe 2-3 seconds long and the connection crashed. and now it's saying TTS is available while everything else is erroring. I saw that you fixed the TTS service being down forever bug, and when I wiped and reinstalled this happened.

`[+] Running 23/23
 ✔ gradio-ui Pulled                                                                                               29.0s
   ✔ 8a47f6aad193 Already exists                                                                                   0.0s
   ✔ a1235d039a7d Already exists                                                                                   0.0s
   ✔ e17464c8c9fb Already exists                                                                                   0.0s
   ✔ f344618db07e Already exists                                                                                   0.0s
   ✔ d6d74d163443 Pull complete                                                                                    4.7s
   ✔ a28c45b5b6fc Pull complete                                                                                   28.3s
   ✔ 89454a8798a7 Pull complete                                                                                   28.3s
   ✔ d114c1fa755e Pull complete                                                                                   28.4s
 ✔ kokoro-tts Pulled                                                                                             109.6s
   ✔ 43f89b94cd7d Pull complete                                                                                    4.0s
   ✔ 45f7ea5367fe Pull complete                                                                                    4.5s
   ✔ 3d97a47c3c73 Pull complete                                                                                    6.3s
   ✔ 12cd4d19752f Pull complete                                                                                    6.3s
   ✔ da5a484f9d74 Pull complete                                                                                    6.3s
   ✔ 46bf552296d3 Pull complete                                                                                   13.5s
   ✔ bc5fbb7dec1e Pull complete                                                                                  102.1s
   ✔ 26ed3e2fc9ef Pull complete                                                                                  102.2s
   ✔ 5982e1357457 Pull complete                                                                                  108.9s
   ✔ 2d47cd31650f Pull complete                                                                                  108.9s
   ✔ c2fdb8f9d6cc Pull complete                                                                                  108.9s
   ✔ a8543fc3149a Pull complete                                                                                  109.0s
   ✔ 7a95b0bb46cb Pull complete                                                                                  109.0s
[+] Running 3/3
 ✔ Container kokoro-fastapi-model-fetcher-1  Created                                                               0.0s
 ✔ Container kokoro-fastapi-kokoro-tts-1     Recreated                                                             0.7s
 ✔ Container kokoro-fastapi-gradio-ui-1      Recreated                                                             0.1s
Attaching to gradio-ui-1, kokoro-tts-1, model-fetcher-1
model-fetcher-1  | Cloning into '.'...
Filtering content: 100% (17/17), 820.18 MiB | 21.72 MiB/s, done.86 MiB/s
kokoro-tts-1     | INFO:     Started server process [1]
kokoro-tts-1     | INFO:     Waiting for application startup.
kokoro-tts-1     | 02:01:34 AM | INFO     | Loading TTS model and voice packs...
kokoro-tts-1     | 02:01:34 AM | INFO     | CUDA available: True
kokoro-tts-1     | 02:01:34 AM | INFO     | CUDA test successful
kokoro-tts-1     | 02:01:34 AM | INFO     | Initializing model on cuda
kokoro-tts-1     | 02:01:34 AM | INFO     | Initializing GPU model
kokoro-tts-1     | /usr/local/lib/python3.10/dist-packages/torch/nn/utils/weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
kokoro-tts-1     |   WeightNorm.apply(module, name, dim)
kokoro-tts-1     | /usr/local/lib/python3.10/dist-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
kokoro-tts-1     |   warnings.warn(
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice af to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice af_bella to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice af_nicole to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice af_sarah to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice af_sky to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice am_adam to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice am_michael to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice bf_emma to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice bf_isabella to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice bm_george to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Copying base voice bm_lewis to voices directory
kokoro-tts-1     | 02:01:37 AM | INFO     | Pre-loaded 1 voices into cache
kokoro-tts-1     | 02:01:37 AM | INFO     | Running warmup inference on voice af
kokoro-tts-1     | 02:01:40 AM | INFO     | Completed warmup for voice af
kokoro-tts-1     | 02:01:40 AM | INFO     | Model warm-up complete
kokoro-tts-1     | 02:01:40 AM | INFO     |
kokoro-tts-1     |
kokoro-tts-1     | ░░░░░░░░░░░░░░░░░░░░░░░░
kokoro-tts-1     |
kokoro-tts-1     |     ╔═╗┌─┐┌─┐┌┬┐
kokoro-tts-1     |     ╠╣ ├─┤└─┐ │
kokoro-tts-1     |     ╚  ┴ ┴└─┘ ┴
kokoro-tts-1     |     ╦╔═┌─┐┬┌─┌─┐
kokoro-tts-1     |     ╠╩╗│ │├┴┐│ │
kokoro-tts-1     |     ╩ ╩└─┘┴ ┴└─┘
kokoro-tts-1     |
kokoro-tts-1     | ░░░░░░░░░░░░░░░░░░░░░░░░
kokoro-tts-1     |
kokoro-tts-1     | Model warmed up on cuda
kokoro-tts-1     | 11 voice packs loaded
kokoro-tts-1     |
kokoro-tts-1     | ░░░░░░░░░░░░░░░░░░░░░░░░
kokoro-tts-1     |
kokoro-tts-1     | INFO:     Application startup complete.
kokoro-tts-1     | INFO:     Uvicorn running on http://0.0.0.0:8880 (Press CTRL+C to quit)
gradio-ui-1      | * Running on local URL:  http://0.0.0.0:7860
gradio-ui-1      |
gradio-ui-1      | To create a public link, set `share=True` in `launch()`.
kokoro-tts-1     | INFO:     172.18.0.4:37520 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:39854 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:39860 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-05-18_voice-af_bella_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:54706 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:54710 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-05-28_voice-af_bella_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:58346 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:58352 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-05-40_voice-af_nicole_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:37204 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:37210 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-05-49_voice-af_sarah_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:37216 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:37232 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-05-54_voice-af_sky_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:57012 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:57018 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-05-59_voice-am_adam_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:57022 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:57030 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-06-04_voice-am_michael_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:57562 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:57574 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-06-08_voice-bf_emma_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:57576 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:57582 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-06-13_voice-bf_isabella_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:40890 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:40892 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-06-18_voice-bm_george_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:40904 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:40920 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-06-23_voice-bm_lewis_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
kokoro-tts-1     | INFO:     172.18.0.4:33228 - "GET /v1/audio/voices HTTP/1.1" 200 OK
kokoro-tts-1     | INFO:     172.18.0.4:33234 - "POST /v1/audio/speech HTTP/1.1" 200 OK
gradio-ui-1      | /usr/local/lib/python3.10/site-packages/gradio/components/dropdown.py:226: UserWarning: The value passed into gr.Dropdown() is not in the list of choices. Please update the list of choices to include: output_2025-01-13_02-06-37_voice-af_nicole_speed-1.mp3 or set allow_custom_value=True.
gradio-ui-1      |   warnings.warn(
Gracefully stopping... (press Ctrl+C again to force)
dependency failed to start: container kokoro-fastapi-kokoro-tts-1 is unhealthy
`

**Comment by pkp24** [comment]
Oh also here is a screenshot
![image](https://github.com/user-attachments/assets/a943d059-8151-4878-aaa3-bcca1ebb5000)

**Comment by pkp24** [comment]
Oh, it's time based. Doesn't matter what I'm doing beforehand. Same prompt didn't crash it, but then after a while it crashed.

**Comment by remsky** [comment]
@pkp24  I'll take a look. Is that when connecting to the cpu, or gpu versions?

**Comment by pkp24** [comment]
It's whatever the default is

**Comment by remsky** [comment]
Should be fixed in the most recent, let me know if it gives you any trouble

https://github.com/remsky/Kokoro-FastAPI/releases/tag/v0.0.5post1

**Comment by pkp24** [comment]
It's fixed, I left it up all day and it hasn't crashed
